### PR TITLE
feat: case url links to tabs (FPLA-2289)

### PR DESCRIPTION
**Pull request by joez-f**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-2289

### Change description ###

When sending sign in links in notifications, direct the user to an appropriate tab based on the email so that they can quickly view relevant documents/information

Scenarios covered
- C2 -> C2 tab
- SDO + CMO + other orders -> orders tab
- Draft CMO -> draft orders tab
- NoH + NoH -> hearings tab
- Placement -> placement tab

Notifications which don't link to tab:
- representative added to case
- judge allocated
- c110a application emails + application returned

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProviderTest.java [hunk 1]
```diff
@@ -28,6 +29,10 @@ protected String caseUrl(String caseId) {
         return formatCaseUrl(UI_URL, Long.valueOf(caseId));
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProviderTest.java:29]: Could just call overloaded method - `return caseUrl(UI_URL, Long.valueOf(caseId), null);`

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProviderTest.java [hunk 2]
```diff
@@ -41,5 +46,8 @@ protected String caseUrl(String caseId) {
     void initCaseUrlService() {
         when(caseUrlService.getCaseUrl(anyLong()))
             .thenAnswer(invocation -> caseUrl(invocation.getArgument(0).toString()));
+
+        when(caseUrlService.getCaseUrl(anyLong(), anyString()))
+            .thenAnswer(invocation -> caseUrl(invocation.getArgument(0).toString(), invocation.getArgument(1)));
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProviderTest.java:51]: Do we need to mock these params?

Comment (joez-f) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProviderTest.java:51]: This is essentially dynamic mocking

**Review by Jamesnelson117 — APPROVED** [review]
LGTM

**Pull request merged**